aws/resource_aws_ec2_local_gateway_route_table_vpc_association.go

```diff
@@ -0,0 +1,174 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/ec2/waiter"
+)
+
+const (
+	// Missing constant in AWS Go SDK
+	ec2ResourceTypeLocalGatewayRouteTableVpcAssociation = "local-gateway-route-table-vpc-association"
+)
+
+func resourceAwsEc2LocalGatewayRouteTableVpcAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsEc2LocalGatewayRouteTableVpcAssociationCreate,
+		Read:   resourceAwsEc2LocalGatewayRouteTableVpcAssociationRead,
+		Update: resourceAwsEc2LocalGatewayRouteTableVpcAssociationUpdate,
+		Delete: resourceAwsEc2LocalGatewayRouteTableVpcAssociationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"local_gateway_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"local_gateway_route_table_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"tags": tagsSchema(),
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsEc2LocalGatewayRouteTableVpcAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	req := &ec2.CreateLocalGatewayRouteTableVpcAssociationInput{
+		LocalGatewayRouteTableId: aws.String(d.Get("local_gateway_route_table_id").(string)),
+		TagSpecifications:        ec2TagSpecificationsFromMap(d.Get("tags").(map[string]interface{}), ec2ResourceTypeLocalGatewayRouteTableVpcAssociation),
+		VpcId:                    aws.String(d.Get("vpc_id").(string)),
+	}
+
+	output, err := conn.CreateLocalGatewayRouteTableVpcAssociation(req)
+
+	if err != nil {
+		return fmt.Errorf("error creating EC2 Local Gateway Route Table VPC Association: %w", err)
+	}
+
+	d.SetId(aws.StringValue(output.LocalGatewayRouteTableVpcAssociation.LocalGatewayRouteTableVpcAssociationId))
+
+	if _, err := waiter.LocalGatewayRouteTableVpcAssociationAssociated(conn, d.Id()); err != nil {
+		return fmt.Errorf("error waiting for EC2 Local Gateway Route Table VPC Association (%s) to associate: %w", d.Id(), err)
+	}
+
+	return resourceAwsEc2LocalGatewayRouteTableVpcAssociationRead(d, meta)
+}
+
+func resourceAwsEc2LocalGatewayRouteTableVpcAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	association, err := getEc2LocalGatewayRouteTableVpcAssociation(conn, d.Id())
```

Comment:
just an aside: could we re-use the `LocalGatewayRouteTableVpcAssociationState` method in the waiter package or would that be a bit unreadable (also in the Exists/Destroy methods of the test file)?

Reply:
Its a decent idea, but a little confusing usage-wise since its conflating two use cases as sometimes the RefreshStateFunc will have special error handling logic that may not be desirable for other uses. I think @gdavison and I are converging on having some form an internal service package that contains consistently named `Get{THING}` functions that the `Read`, `testAccCheck{THING}Exists`, `testAccCheck{THING}Destroy`, and the waiter package functions will all commonly use. 👍